packages/@aws-cdk/aws-apigateway/lib/restapi.ts

```diff
@@ -159,6 +159,14 @@ export interface RestApiOptions extends ResourceOptions {
    * @default - when no export name is given, output will be created without export
    */
   readonly endpointExportName?: string;
+
+  /**
+   * The EndpointConfiguration property type specifies the endpoint types of a REST API
+   * @see https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-apigateway-restapi-endpointconfiguration.html
+   *
+   * @default - No endpoint configuration
+   */
+  readonly endpointConfiguration?: EndpointConfiguration;
```

Comment:
[`EndpointConfiguration`](https://docs.aws.amazon.com/cdk/api/latest/docs/@aws-cdk_aws-apigateway.EndpointConfiguration.html#vpcendpoints) also includes VPC endpoints which conflicts with https://docs.aws.amazon.com/apigateway/latest/developerguide/api-gateway-swagger-extensions-endpoint-configuration.html.

It's better only support properties that cannot be specified via the OpenAPI spec. 

If you only need this for specifying the types, maybe we can use `EndpointType` here directly.

Reply:
addressed in https://github.com/aws/aws-cdk/pull/9068/commits/7a097a16e6aac372d2485790807310ef9df8633a
I have undeprecated that method and move it into the RestApiOptions (and moved back the endpointConfiguration)